public/js/actions/sources.js

```diff
@@ -87,6 +83,16 @@ function newSource(source) {
   };
 }
 
+function newSources(sources) {
+  return ({ dispatch, getState }) => {
+    sources.filter(source => {
+      return !getSource(getState(), source.id);
+    }).forEach(source => {
+      dispatch(newSource(source));
+    });
+  };
```

Comment:
the most minor of nits, but maybe inline these so that it's a little bit more compact.

```js
sources.filter(source => !getSource(getState(), source.id))
             .forEach(source => dispatch(newSource(source)));
```